examples/serialization/mixed-type-collection/main.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This example shows how to use mixed type collections. It can also be used to
+ * prevent de-serialization and do late de-serialization. You can query the type
+ * information of a value during runtime and take action based on the type.
+ */
+
+#include <hazelcast/client/hazelcast_client.h>
+#include <ostream>
+
+int
+main()
+{
+    using namespace hazelcast::client;
+
+    auto hz = hazelcast::new_client().get();
+
+    auto map = hz.get_map("map").get();
+
+    boost::optional<typed_data> the_value =
```

Comment:
`typed_data` corresponds to Java client's `Data`, doesn't it? I thought that was private API.

Reply:
No, there is another Data type which is internal. This was supposed to be public API. 

The package name is incorrect but it was released like this for long ago. I like to deprecate and move it to package `hazelcast::client::serialization` rather than `pimpl` which is normally implementation namespace.